rust/agama-server/src/cert.rs

```diff
@@ -0,0 +1,63 @@
+use openssl::error::ErrorStack;
+use openssl::pkey::{PKey, Private};
+use openssl::rsa::Rsa;
+use openssl::x509::extension::{KeyUsage, SubjectAlternativeName};
+use openssl::x509::{X509NameBuilder, X509};
+
```

Comment:
Generating a self-signed certificate manually [is not trivial](https://stackoverflow.com/questions/10175812/how-to-generate-a-self-signed-ssl-certificate-using-openssl).

I thought it would be a good idea to reuse the code for some separate tool like `agama-gen-cert`. You would call `agama-gen-cert mycert.pem mykey.pem` to obtain your certificate easily. (Probably with few optional configuration settings like `--hostname myagama.mydomain`.)

Then you would just copy them to a running Agama instance.

Reply:
Sorry, I meant the blank-line 😅